inspirehep/records/marshmallow/literature/base.py

```diff
@@ -124,3 +129,163 @@ def get_ui_display(self, data):
             return json.loads(ui_display)
         except json.JSONDecodeError:
             return {}
+
+
+class LiteratureESEnhancementV1(LiteratureMetadataSchemaV1):
+    """Elasticsearch serialzier"""
+
+    _created = fields.DateTime(dump_only=True, attribute="created")
+    _updated = fields.DateTime(dump_only=True, attribute="updated")
+    abstract_source_suggest = fields.Method("populate_abstract_source_suggest")
+    author_count = fields.Method("populate_author_count")
+    authors = fields.Method("preprocess_authors")
+    bookautocomplete = fields.Method("populate_bookautocomplete")
+    earliest_date = fields.Method("populate_earliest_date")
+    facet_inspire_doc_type = fields.Method("populate_inspire_document_type")
+    facet_author_name = fields.Method("populate_facet_author_name")
+    id = fields.UUID(dump_only=True)
+    thesis_info = fields.Nested(ThesisInfoSchemaForESV1, dump_only=True)
+
+    def populate_abstract_source_suggest(self, record):
+        """Prepares record for ``abstract_source_suggest`` field."""
+        abstracts = record.get("abstracts", [])
+
+        for abstract in abstracts:
```

Comment:
it's not OK to go nested inside `field.Method`, you should create a new schema instead. For example for this case:

`AbstractSchema` with all fields `fields.Raw` and `abstract_source_suggest = fields.Method()` and then use it up there `abstracts: fields.Nested(AbstractSchema, many=True)`